### ISSUE TITLE
Optional steno keyboard overlay 

### ISSUE THREAD

**Issue by adiron** [openstenoproject/plover#646]
Classification: Feature (New)

Reproducibility: Not Applicable - N/A 

## Summary

When first learning steno, it's sometimes very hard to remember which key corresponds to which letter. As "training wheels" for new users, it might be beneficial to have a virtual steno keyboard that shows the keys being pressed as they are pressed on the input device.

This overlay would help in the following ways:
1. Help memorize the key positions and hopefully develop muscle memory.
2. Highlight depressed keys, and therefor help the user spot mistakes as they are typed.

## Notes

As with all things "training wheels" there's a discussion about whether it contributes to learning or slows it down. Personally, I think that with something with as steep a learning curve as steno, there probably needs to be something to lower the requirements for entry. Since the overlay would take up screen real-estate, I imagine that users will eventually want to turn this off after they get the hang steno.

**Comment by GuiltyDolphin** [comment]
@adiron What's wrong with the existing stroke (and suggestion) displays? The stroke display will show you which steno keys are being pressed (in the correct order and position), while the suggestion display will show what translations result from the keys (also shows which keys are pressed).

**Comment by adiron** [comment]
I don't think so much that there's anything "wrong" with the stroke display, but I think that it can be improved, and that a lot can be said for visual learning aids. It could be useful when demonstrating or teaching steno or when learning individually.

I don't personally think it should display a QWERTY keyboard but rather a steno keyboard (even if using QWERTY-based input). Similar to what happens [on this site when you click "always show hints"](http://qwertysteno.com/Intermediate/WordBeginningsPractice.php).

**Comment by GuiltyDolphin** [comment]
@adiron I quite like the idea of using it in demonstrations ("Your thumb rests across these keys and corresponds to the vowels like so..."), and it might be quite good for debugging positions when learning.

Do you think this should be another option for the stroke display? There are already a couple of options so it seems a fairly logical place.

**Comment by adiron** [comment]
@GuiltyDolphin I agree that the stroke window is a good place place for it. Maybe also with a "pop-out" presentation mode that will make it transparent, floating and without any window decorations.

**Comment by morinted** [comment]
Benoit Pierre had implemented something like this a while back to offer a
visualization of the keymap editor.

Note that the stroke would only get sent to Plover (and the GUI, for
display purposes) after it has been released. Does that make sense?

**Comment by nsmarkop** [comment]
Does the Plover Layout Display plugin meet this need? https://github.com/morinted/plover_layout_display It has options for adding key labels and whatever layout you want now to visualize input.

**Comment by memoriasIT** [comment]
> 
> 
> Does the Plover Layout Display plugin meet this need? https://github.com/morinted/plover_layout_display It has options for adding key labels and whatever layout you want now to visualize input.

While that plugin is great, it does not show key presses in real time, but last input.
Having an option to make it real time could be beneficial so you don't have to be looking at your keyboard and comparing to the layout.

**Comment by morinted** [comment]
Realtime would require tighter integration with the machine protocol than
Plover exposes. Additionally, this information is only available in select
protocols -- namely "Keyboard" and "Treal"
